torch/csrc/jit/passes/dead_code_elimination.cpp

```diff
@@ -6,7 +6,7 @@ void EliminateDeadCode(std::shared_ptr<Graph>& graph) {
   EliminateDeadCode(graph->block());
 }
 bool hasSideEffects(Node * node) {
-  return node->kind() == prim::Print;
+  return node->kind() == prim::Print || node->blocks().size() > 0;
```

Comment:
Why is this needed? Because we can reference values implicitly from blocks?

Reply:
I think previously we could have removed an `If` node that has no outputs (but its body can have side effects!).

Reply:
Correct. I hit this while debugging and saw my print inside and if disappear. We really should change to the version @ezyang suggested where side-effecting operators like print take and return a "world" token. which would then become an output of the if-statement and then we can remove the `hasSideEffects` check.